src/main/java/io/temporal/internal/statemachines/ActivityStateMachine.puml

```diff
@@ -0,0 +1,54 @@
+` Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+`
+` Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+`
+` Modifications copyright (C) 2017 Uber Technologies, Inc.
+`
+` Licensed under the Apache License, Version 2.0 (the "License"). You may not
+` use this file except in compliance with the License. A copy of the License is
+` located at
+`
+` http://aws.amazon.com/apache2.0
+`
+` or in the "license" file accompanying this file. This file is distributed on
+` an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+` express or implied. See the License for the specific language governing
+` permissions and limitations under the License.
+
+` PlantUML <plantuml.com> State Diagram.
+` Generated from src/main/java/io/temporal/internal/statemachines/ActivityStateMachine.java
+` by io.temporal.internal.statemachines.CommandsGeneratePlantUMLStateDiagrams.
+
+
+@startuml
+scale 350 width
+[*] --> CREATED
+CREATED --> SCHEDULE_COMMAND_CREATED: SCHEDULE
+SCHEDULE_COMMAND_CREATED --> SCHEDULE_COMMAND_CREATED: SCHEDULE_ACTIVITY_TASK
+SCHEDULE_COMMAND_CREATED --> SCHEDULED_EVENT_RECORDED: ACTIVITY_TASK_SCHEDULED
+SCHEDULE_COMMAND_CREATED --> CANCELED: CANCEL
+SCHEDULED_EVENT_RECORDED --> STARTED: ACTIVITY_TASK_STARTED
+SCHEDULED_EVENT_RECORDED --> TIMED_OUT: ACTIVITY_TASK_TIMED_OUT
+STARTED --> COMPLETED: ACTIVITY_TASK_COMPLETED
+STARTED --> FAILED: ACTIVITY_TASK_FAILED
+STARTED --> TIMED_OUT: ACTIVITY_TASK_TIMED_OUT
+SCHEDULED_EVENT_RECORDED --> SCHEDULED_ACTIVITY_CANCEL_COMMAND_CREATED: CANCEL
+SCHEDULED_ACTIVITY_CANCEL_COMMAND_CREATED --> SCHEDULED_ACTIVITY_CANCEL_EVENT_RECORDED: ACTIVITY_TASK_CANCEL_REQUESTED
+SCHEDULED_ACTIVITY_CANCEL_COMMAND_CREATED --> SCHEDULED_ACTIVITY_CANCEL_COMMAND_CREATED: REQUEST_CANCEL_ACTIVITY_TASK
+SCHEDULED_ACTIVITY_CANCEL_COMMAND_CREATED --> TIMED_OUT: ACTIVITY_TASK_TIMED_OUT
+SCHEDULED_ACTIVITY_CANCEL_EVENT_RECORDED --> CANCELED: ACTIVITY_TASK_CANCELED
+SCHEDULED_ACTIVITY_CANCEL_EVENT_RECORDED --> STARTED_ACTIVITY_CANCEL_EVENT_RECORDED: ACTIVITY_TASK_STARTED
+STARTED_ACTIVITY_CANCEL_COMMAND_CREATED --> STARTED_ACTIVITY_CANCEL_EVENT_RECORDED: ACTIVITY_TASK_CANCEL_REQUESTED
+STARTED_ACTIVITY_CANCEL_COMMAND_CREATED --> STARTED_ACTIVITY_CANCEL_COMMAND_CREATED: REQUEST_CANCEL_ACTIVITY_TASK
+STARTED --> STARTED_ACTIVITY_CANCEL_COMMAND_CREATED: CANCEL
```

Comment:
Same comment as above file. Can we sort by source state?

Reply:
Done